### PR TITLE
Call VSP to fetch locations info

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This updates the new appointment flow to call VSP to get Location data. This happens through the HealthcareService service, though I haven't incorporated the HealthcareService resources into the data used by the new appointment flow, since that's a fair amount of work and we don't know for sure how VSP will handle that.

Most of the new lines in here are mock data I copied from staging.

## Testing done
Local and unit testing

## Acceptance criteria
- [ ] VSP is called to fetch location list via healthcare services

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/actions/newAppointment.js [hunk 1]
```diff
@@ -295,27 +293,30 @@ export function openFacilityPage(page, uiSchema, schema) {
...
 
-      facilities =
+      locations =
         newAppointment.facilities[`${typeOfCareId}_${parentId}`] || null;
 
-      if (canShowFacilities && !facilities) {
-        facilities = await getSupportedLocationsByTypeOfCare({
-          rootOrgId,
+      if (canShowFacilities && !locations) {
+        ({ locations } = await getSupportedHealthcareServicesAndLocations({
```
Comment (narin) [inline src/applications/vaos/actions/newAppointment.js:305]: What's the purpose of the parentheses surrounding this statement?

File: src/applications/vaos/utils/eligibility.js [hunk 2]
```diff
@@ -99,9 +100,9 @@ export async function getEligibilityData(
   const eligibility = {
     ...results,
     hasMatchingClinics: !!results.clinics?.length,
-    directSupported: location.legacyVAR.directSchedulingSupported,
+    directSupported: useVSP || location.legacyVAR.directSchedulingSupported,
```
Comment (narin) [inline src/applications/vaos/utils/eligibility.js:103]: Does this imply that direct scheduling is always supported with VSP?

**Review by narin — APPROVED** [review]
LGTM but just a couple questions for understanding.

File: src/applications/vaos/actions/newAppointment.js (hunk 1 above)
Comment (jbalboni) [reply src/applications/vaos/actions/newAppointment.js:305]: They're required if you're destructuring into a variable that's already been defined. It's weird and annoying.

File: src/applications/vaos/utils/eligibility.js (hunk 2 above)
Comment (jbalboni) [reply src/applications/vaos/utils/eligibility.js:103]: Yes, for now. I figured it was preferable to be able to test the VSP flow before all the eligibility checks get implemented in VSP, and I didn't want to add the `legacyVAR` object to data that actually came from VSP.

**Pull request merged**